utility_scripts/tests/test_balance_dataset.py

```diff
@@ -0,0 +1,488 @@
+        cls.dataset = pd.read_csv(
+            os.path.join(
+                cls.__current_directory__,
+                'utility_scripts',
+                'tests',
+                'resources',
+                'train_input.tsv.gz'
+            ),
+            sep='\t',
+            na_values='.',
+            low_memory=False
+        )
+        cls.hardcoded_columns = ['variant', 'Consequence', 'gnomAD_AF', 'binarized_label']
+        cls.test_sampler_data = pd.DataFrame(
+            {
+                'variant': ['variant_1', 'consequence_1', 0.02, 0],
+                'Consequence': ['variant_2', 'consequence_1', 0.01, 1],
+                'gnomAD_AF': ['variant_3', 'consequence_2', 0.02, 0],
+                'binarized_label': ['variant_4', 'consequence_2', 0.01, 1]
+            }
+        )
+
+    @classmethod
+    def tearDownClass(cls) -> None:
+        if os.path.isdir(cls.depth_1_directory):
+            os.rmdir(cls.depth_1_directory)
+        if os.path.isdir(cls.depth_2_directory):
+            os.rmdir(cls.depth_2_directory)
+            os.rmdir(cls.depth_2_directory.parent)
+        if os.path.isdir(cls.__test_path__):
+            for filename in os.listdir(cls.__test_path__):
+                filepath = os.path.join(cls.__test_path__, filename)
+                try:
+                    if os.path.isfile(filepath) or os.path.islink(filepath):
+                        os.unlink(filepath)
+                    elif os.path.isdir(filepath):
+                        shutil.rmtree(filepath)
+                except Exception as e:
+                    print(f'Failed to delete {filepath}, reason: {e}')
+            try:
+                os.rmdir(cls.__test_path__)
+            except Exception as e:
+                print(f'Failed to remove tests folder {cls.__test_path__}, reason: {e}')
+
+    def setUp(self) -> None:
+        print('Testing Class:')
+
+    def tearDown(self) -> None:
+        print('Done.')
+
+    def test_obtain_consequences(self):
+        balancer = Balancer(VerbosityPrinter())
+        consequence_series = pd.Series(
+            name='Consequence',
+            data=[
+                'consequence_1',
+                'consequence_2',
+                'consequence_3&consequence_4',
+                'consequence_3'
+            ]
+        )
+        self.assertTrue(
+            np.array_equal(
+                np.array(
+                    [
+                        'consequence_1',
+                        'consequence_2',
+                        'consequence_3',
+                        'consequence_4'
+                    ]
+                ),
+                balancer._obtain_consequences(consequence_series)
+            )
+        )
+
+    def test_sampler_unchanged_more_required(self):
+        balancer = Balancer(VerbosityPrinter())
+        test_set = self.test_sampler_data.copy(deep=True)
+        self.assertEqual(balancer._sample_variants(test_set, 5).shape[0], 4)
```

Comment:
Yes this is intended, else every single consequence would throw an error unless the amount of pathogenic and benign are exactly equal (for the allele frequency bin), since I perform the check for both benign and pathogenic.

Reply:
With the refactor in mind maybe not worth it to change it here, but I would suggest renaming it in the refactor to something else as the name seems confusing. For example: `n_max` or `reduce_to`.